compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -352,6 +353,63 @@ pub enum DocStyle {
     Inner,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct LocatedToken(Located<Token>);
+
+impl PartialEq<LocatedToken> for Token {
+    fn eq(&self, other: &LocatedToken) -> bool {
+        self == &other.0.contents
+    }
+}
+impl PartialEq<Token> for LocatedToken {
+    fn eq(&self, other: &Token) -> bool {
+        &self.0.contents == other
+    }
+}
+
+impl From<LocatedToken> for Token {
+    fn from(spt: LocatedToken) -> Self {
+        spt.0.contents
+    }
+}
+
+impl<'a> From<&'a LocatedToken> for &'a Token {
+    fn from(spt: &'a LocatedToken) -> Self {
+        &spt.0.contents
+    }
+}
+
+impl LocatedToken {
+    pub fn new(token: Token, location: Location) -> LocatedToken {
+        LocatedToken(Located::from(location, token))
+    }
+    pub fn to_location(&self) -> Location {
```

Comment:
Why `to_location` and not `location` as used elsewhere? Same for `to_span`

Reply:
Oh, it was called `to_span` in `SpannedToken` so I called it `to_location` in `LocatedToken`, just to follow a tradition. I'll change the names.

Reply:
Actually, let me know if you'd prefer me to do this change now or in a follow-up PR.

Reply:
Done!